gax-java/pom.xml

```diff
@@ -222,6 +222,7 @@
         <artifactId>maven-jar-plugin</artifactId>
         <executions>
           <execution>
+            <phase>test</phase>
```

Comment:
Could you clarify what this is doing?

@blakeli0 , @suztomo : Is it possible that the `<skip>true</skip>` three lines down from here is the reason why we have to execute `mvn install` rather than `mvn test` to avoid linking errors?

Reply:
Sure! This line helps resolve the compilation error we see when running `mvn test` at the root. The [PR description](https://github.com/googleapis/gapic-generator-java/pull/1430#issue-1607483042) goes over this briefly but upon comparing the `mvn test` with `mvn install`, I noticed that the `jar:test-jar` step which creates the `testlib` JAR that gapic-generator-java depends on was being executed when `mvn install` was called but **after** `surefire:test` step. On the other hand, it was not being invoked at all when `mvn test` was executed. This change forces the maven-jar-plugin of GAX to be executed at the `test` phase.

Reply:
@blakeli0 and @suztomo for visibility. Great work!